Automattic/vip-design-system
# Footer: Create new component

## Description

Closes: https://github.com/Automattic/vip-design-system/issues/373

We need the `<Footer/>` from the dashboard login page for another app. Instead of recreating it, we can make a reusable component to use in both places. It's based on new designs: 

**Mobile**

<img width="396" alt="Screenshot 2024-06-04 at 1 03 38 PM" src="https://github.com/Automattic/vip-design-system/assets/35543432/0f20fedc-71b9-412e-9785-72fa2e6e3272">

**Desktop**

<img width="1062" alt="Screenshot 2024-06-04 at 1 03 22 PM" src="https://github.com/Automattic/vip-design-system/assets/35543432/6ddf4b99-9147-4e3b-bbbb-75b7084d35ba">

I thought it might be best to keep it as minimal as possible, so it's easy to use and keep consistent. We could always expose more props in the future, but it's not as easy taking them away 😅 

I also considered having an option for `LinkExternal` vs `Link` but then that opens it up to having the button variants (unless we don't allow it) which would require more style changes. And since LinkExternal props can be added, the arrow and screenreader text can be customized. 

## Checklist


## Steps to Test

1. Pull down PR
2. `npm run dev`
3. Open Storybook
4. Navigate to 'Footer' component
5. Ensure links work as expected, component is responsive and accessible
